Client/Frontend/Login/LoginInfoViewController.swift

```diff
@@ -279,33 +313,31 @@ extension LoginInfoViewController {
     }
     
     @objc private func done() {
-        updateLoginInfo { [weak self] in
+        updateLoginInfo { [weak self] success in
             guard let self = self else { return }
 
             self.isEditingFieldData = false
-            self.tableView.reloadData()
+            if success {
+                self.tableView.reloadData()
+            }
             self.navigationItem.rightBarButtonItem =
                 UIBarButtonItem(barButtonSystemItem: .edit, target: self, action: #selector(self.edit))
         }
     }
     
-    private func updateLoginInfo(completion: @escaping () -> Void) {
+    private func updateLoginInfo(completion: @escaping (Bool) -> Void) {
         guard let username = usernameField?.text, let password = passwordField?.text,
-              username != loginEntry.username || password != loginEntry.password else {
-            completion()
+              username != credentials.usernameValue || password != credentials.passwordValue else {
+            completion(false)
             return
         }
-                
-        loginEntry.update(password: password, username: username)
         
-        profile.logins.updateLoginByGUID(loginEntry.guid, new: loginEntry, significant: true).upon { result in
-            DispatchQueue.main.async {
-                if !result.isSuccess {
-                    log.error("Error while updating a login entry. Error Reason: \(result.failureValue ?? "")")
-                }
-                completion()
-            }
+        if let oldCredentials = credentials.copy() as? PasswordForm {
+            credentials.update(username, passwordValue: password)
+            passwordAPI.updateLogin(credentials, oldPasswordForm: oldCredentials)
```

Comment:
You can just force unwrap this, should theoretically always be itself.  If you are going to keep the conditional, you should call `completion` based on whether or not the cast is successful no? Otherwise you don't actually update any logins

Reply:
Fix at https://github.com/brave/brave-ios/pull/5013/commits/1a6d7cfd93a42bc6c37576fe5ddbca88aba7fa7c